lxd/lxd-metadata/lxd_metadata.go

```diff
@@ -285,7 +253,7 @@ func parse(path string, outputJSONPath string, excludedPaths []string, substitut
 						continue
 					}
 
-					configKeyEntry[metadataMap["key"]].(map[string]any)[dataKVMatch[1]] = detectType(dataKVMatch[2])
+					configKeyEntry[metadataMap["key"]].(map[string]any)[dataKVMatch[1]] = dataKVMatch[2]
```

Comment:
I'm not sure about this change.

Where is the value in `dataKVMatch[2]` coming from?

As for how these are classified, whilst I agree that all config values are strings, my understanding of the metadata definition here is in order to indicate in the docs what sort of string values are allowed for a particular config key.

Reply:
It looks like `dataKVMatch` is coming from this regular expression `(?m)([\S]+):[\s]+([\S \"\']+)` on the key value section of the `lxdmeta:generate` section of the doc block e.g.:
```
type: string
defaultdesc: `auto`
liveupdate: no
shortdesc: What to do when evacuating the instance
```
In this example, the first `dataKVMatch` outputted from `lxdDocDataRegex.FindAllStringSubmatch(data, -1)` would have
```
dataKVMatch[1] -> "type"
dataKVMatch[2] -> "string"
```

My feeling is that lxd-generate shouldn't be too clever about figuring out which config keys are which type. We should just include that in the description of the key. There is only one config key for which this appears to work.

Reply:
>  There is only one config key for which this appears to work.
which one?

Reply:
The commit directly after this change shows the result of `make update-metadata`. Only one field is changed: https://github.com/canonical/lxd/pull/14132/commits/2614bb5bba2ea25288f0f2b391f70c6eab7843e3#diff-d3459341f992192c3b18b6130aa7e165d305183794a84f219c7ab7512b1b31bcL4316

It is changing the value of `defaultdesc` from a boolean `true` to a string `"true"`.